### ISSUE TITLE
required, nullable class variables are not generated as nullable

### ISSUE THREAD

**Issue by NANASHI0X74** [dart-ogurets/dart-openapi-maven#58]
example schema:
```yaml
components:
  schemas:
    UserDetail:
      type:object
      properties:
        invite:
          type: boolean
          readOnly: true
          nullable: true
      required:
      - invite
```

class generated will have invite as `late bool invite` and throw a deserializationerror when deserializing objects that have that property as null.

**Comment by rvowles** [comment]
Yeah, so this is me ignoring that completely, always. Its going to require a full sweep.

**Comment by gbloggs** [comment]
also, a few others:

```
        <groupId>org.openapitools</groupId>
        <artifactId>openapi-generator-maven-plugin</artifactId>
        <version>5.2.0</version>

            <groupId>com.bluetrainsoftware.maven</groupId>
            <artifactId>openapi-dart-generator</artifactId>
            <version>5.8</version>
```

This optional in a paged response:
```
      parameters:
        - $ref: '#/components/parameters/start-date'
        - $ref: '#/components/parameters/end-date'
        - $ref: '#/components/parameters/page'
        - $ref: '#/components/parameters/limit'
        - $ref: '#/components/parameters/sort'
        ...
    start-date:
      name: start-date
      in: query
      description: the start date to apply to the request
      required: false
      schema:
        type: string
        format: date
        maxLength: 10
        example: '2020-06-30'
    page:
      name: page
      in: query
      description: Current page number
      required: false
      schema:
        type: integer
        format: int32
        default: 0
        minimum: 0
        maximum: 32768
    limit:
      name: limit
      in: query
      description: Limits the number of returned items
      required: false
      schema:
        type: integer
        format: int32
        default: 20
        minimum: 10
        maximum: 1000
        multipleOf: 10
```

interesting that it was only limit...

```
  LocalApiClient.parameterToString(p),
          '', 'limit', limit));
          }
```
Needed to be:
```
  LocalApiClient.parameterToString(p)!,
          '', 'limit', limit));
          }
```

Also, all the:

```
openApiDateTimeParameterToString(p)!,
          '', 'start-date-time', startDateTime));
          }
```
and 
```
      openApiDateParameterToString(p)!,
          '', 'start-date', startDate));
          }
```
had to have the ! removed:
```
openApiDateTimeParameterToString(p),
          '', 'start-date-time', startDateTime));
          }

...

      openApiDateParameterToString(p),
          '', 'start-date', startDate));
          }
```

**Comment by rvowles** [comment]
nullable: true is going away in the OpenAPI spec, and this is why i have ignored it for the 5.x series. OpenAPI is going into 6.x so I will re-look at what they have done there, but it will generate incompatible code for a lot of people I expect.

@gbloggs that is a different thing - the original issue is that a required field is one that must be sent, but it can have a null value. I will investigate that as a separate ticket.

**Comment by rvowles** [comment]
I think I will be able to solve these two issues actually.

**Comment by rvowles** [comment]
these are resolved in the 6.1 release

**Comment by gbloggs** [comment]
Cheers Richard. Will throw everything through the mixer over the weekend
________________________________
